src/Storages/MergeTree/MergeTreeDataPartWriterWide.cpp

```diff
@@ -126,21 +134,30 @@ void MergeTreeDataPartWriterWide::addStreams(
             marks_compression_codec,
             settings.marks_compress_block_size,
             settings.query_write_settings);
+
+        full_name_to_stream_name.emplace(full_stream_name, stream_name);
+        stream_name_to_full_name.emplace(stream_name, full_stream_name);
     };
 
     ISerialization::SubstreamPath path;
     data_part->getSerialization(column.name)->enumerateStreams(callback, column.type);
 }
 
+const String & MergeTreeDataPartWriterWide::getStreamName(
+    const NameAndTypePair & column,
+    const ISerialization::SubstreamPath & substream_path) const
+{
+    auto full_stream_name = ISerialization::getFileNameForStream(column, substream_path);
+    return full_name_to_stream_name.at(full_stream_name);
```

Comment:
It's too easy to forget to call it and tests will not catch it

Reply:
Unfortunately in writer there is no other way to do it.